### ISSUE TITLE
Install pnpm after validating node and npm version

### ISSUE THREAD

**Issue by mehdi-vaadin** [vaadin/flow#7020]
Part of #6966 
Depends on #7019 

With setting a configuration key, e.g. `disable.pnpm`, the user should be able to switch to `npm`; in that case, `pnpm` doesn't need to be installed.

**Comment by joheriks** [comment]
Acceptance criteria:

- After implementation, `pnpm`is automaticlly installed as a local module
- The parameter `disable.pnpm` should be available both as a Maven parameter and a Vaadin configuration property

**Comment by denis-anisimov** [comment]
This ticket will be in fact superseded by the #7021
Validation which is mentioned in the description is a bad place to install `pnpm` for many reasons:
- validation in the plugin is done inside `prepare-frontend` goal. This is a bad place to install `pnpm` since the goal is used in both dev and production mode. If you are using dev mode then you will need to disable `pnpm` twice: in the `prepare-frontend`  via config parameter and in the runtime via deployment config parameter. The problem is : maven config parameter can't contain dots and it never matches the servlet init parameter. You may use a system property for both cases. But system property for maven plugin will be `disable.pnpm`  and `vaadin.disable.pnpm` for servlet (and it's `disable.pnpm` as an init parameter property). As a result it's strongly inconvenient to use two properties to disable one feature. We may decide later on to install `pnpm` in `prepare-frontend`   for the sake of speed but this should be a totally separate consideration and a ticket.
- It's quite artificial to install `pnpm` without a need: `pnpm` should be installed at the first time when it's needed. And since this ticket is not about `pnpm` usage it can't be installed without usage. Again: later on we may decide to install it in advance to speed up things. But the above consideration should be taken into account. This should be a separate task.
